src/extended/linux-connection-info.js

```diff
@@ -3,43 +3,24 @@ var networkUtils = require('../network-utils');
 var env = require('../env');
 
 function getConnectionInfo(config, connection, callback) {
+  // TODO: This first nmcli command appears to not do anything useful for us in particular? Remove?
+  var commandStr = "nmcli -f ipv4.addresses,ipv4.gateway con show '" + connection +
+    "' | awk '{print $2}' && nmcli -f IP4.ADDRESS,IP4.GATEWAY con show '" + connection +
+    "' | awk '{print $2}'"
 
-  var commandStr = "nmcli -f ipv4.addresses,ipv4.gateway con show " + connection + 
-  " | awk '{print $2}' && nmcli -f IP4.ADDRESS,IP4.GATEWAY con show " + connection + 
-  " | awk '{print $2}'"
-
+  exec(commandStr, env, function (err, stdOut) {
+    if (err) {
+      callback && callback(err);
+      return;
+    }
 
-  exec(commandStr, env, function(err, stdOut) {
-      if (err) {
-        callback && callback(err);
-        return;
-      }
-      var data = stdOut.split('\n');
-      data.pop();
-      var connectionInfo;
-      if (data.length === 2) {
-        if (data[0] === '') {
-          connectionInfo = {
-            address: '',
-            subnet: '',
-            gateway: ''
-          }
-        } else {
-          connectionInfo = {
-            address: data[0].split('/')[0],
-            subnet: data[0].split('/')[1],
-            gateway: data[1]
-          }
-        }
-      } else if (data.length === 4) {
-        connectionInfo = {
-          address: data[2].split('/')[0],
-          subnet: data[2].split('/')[1],
-          gateway: data[3]
-        }
-      }
+    var data = stdOut.split('\n');
```

Comment:
dear lord that previous logic was trash

Reply:
we really should remove that first nmcli call because its not used, but i figured i'd wait to make sure i didn't blow anything else up with this first.